--- src/lib.rs
+++ src/lib.rs
@@ -401,9 +401,15 @@

                             let is_inside = intersections_to_right % 2 != 0;

-                            let distance =
-                                self.mesh.distance_to_local_point(&point_3d, false) as f32;
-                            let sdf = if is_inside { -distance } else { distance };
+                            let sdf = if narrow_band.is_none() {
+                                if is_inside {
+                                    -1.0
+                                } else {
+                                    1.0
+                                }
+                            } else {
+                                let distance = self.mesh.distance_to_local_point(&point_3d, false) as f32;
+                                if is_inside { -distance } else { distance }
+                            };

                             let keep = if let Some(band) = narrow_band {
                                 sdf.abs() <= band as f32
