--- src/lib.rs
+++ src/lib.rs
@@ -161,21 +161,21 @@

         // We avoid collecting the entire yz cartesian product to save memory.
         // Instead we can use rayon's `into_par_iter` on a range or use flat_map across the ranges.
-        let particles: Vec<ParticleData> = (0..ny)
+        let particles: Vec<ParticleData> = (0..nz)
             .into_par_iter()
-            .flat_map(|iy| {
-                (0..nz).into_par_iter().flat_map(move |iz| {
+            .flat_map(|iz| {
+                (0..ny).into_par_iter().flat_map(move |iy| {
                     let mut local_particles = Vec::with_capacity(nx as usize);
                     let y = self.bounds_min.y + (iy as f64 * resolution) + (resolution * 0.5);
                     let z = self.bounds_min.z + (iz as f64 * resolution) + (resolution * 0.5);

                     // Iterate over X in the inner loop to optimize spatial cache locality.
                     // Because rays are cast along the +X direction, doing X sequentially
                     // keeps the raycast traversals in the same BVH region,
                     // while parallelizing over (Y, Z) ensures finer granularity for Rayon.
                     for ix in 0..nx {
                         let x = self.bounds_min.x + (ix as f64 * resolution) + (resolution * 0.5);
                         let point = Point::new(x, y, z);
